src/ml_board/frontend/Dashboard/src/app/App.tsx

```diff
@@ -99,80 +101,82 @@ export default function App() {
-                // update the Charts Slice
-                dispatch(saveEvalResultData(data.evaluationResultsData));
-                // save the latest metric in the Experiment Slice
-                const { epoch, experiment_id, metric_scores, loss_scores } = data.latest_split_metric as EvaluationResultPayload;
-                const changes: { [latest_split_metric_key: string]: number } = {};
-                for (const metric of metric_scores) {
-                    changes[metric.split + "_" + metric.metric] = metric.score;
-                }
-                for (const loss of loss_scores) {
-                    changes[loss.split + "_" + loss.loss] = loss.score;
-                }
-                //NOTE, I checked the epoch against the experiment's and that didn't work because of UseAppSelector! (can't be used here!)
-                dispatch(updateExperiment({ id: experiment_id, changes: changes }))
-            }
-            else if (data && data.jobStatusData) {
-                dispatch(upsertJob(data.jobStatusData))
-            }
-            else if (data && data.experimentStatusData) {
-                dispatch(upsertExperiment(data.experimentStatusData))
-            }
-            else if (data && data.status) {
-                if (data.status === "msg_count_increment") {
-                    dispatch(incrementReceivedMsgCount())
-                } else if (data.status["ping"] !== undefined) {
-                    dispatch(setLastPing(data.status["ping"]))
-                } else if (data.status["throughput"] !== undefined) {
-                    dispatch(setThroughput(data.status["throughput"]))
-                } else if (data.status["isSocketConnected"] !== undefined) {
-                    dispatch(setSocketConnection(data.status["isSocketConnected"]))
-                }
+        else if (data && data.jobStatusData) {
+            dispatch(upsertJob(data.jobStatusData));
+        }
+        else if (data && data.experimentStatusData) {
+            dispatch(upsertExperiment(data.experimentStatusData));
+        }
+        else if (data && data.status) {
+            if (data.status === "msg_count_increment") {
+                dispatch(incrementReceivedMsgCount());
+            } else if (data.status["ping"] !== undefined) {
+                dispatch(setLastPing(data.status["ping"]));
+            } else if (data.status["throughput"] !== undefined) {
+                dispatch(setThroughput(data.status["throughput"]));
+            } else if (data.status["isSocketConnected"] !== undefined) {
+                dispatch(setSocketConnection(data.status["isSocketConnected"]));
+                setConnectionSnackBar({
+                    isOpen: true,
+                    connection: data.status["isSocketConnected"]
+                });
+                // setSnackBarText(`Socket Connection ${data.status["isSocketConnected"] ? "Successful" : "Failed"}`);
+                // setSnackbarOpen(true);
             }
```

Comment:
from a software engineering I would split this up into multiple functions (one for each message type)
```js
if  (data && data.evaluationResultsData) {
processEvaluationResultData(data)
}
else if (data && data.jobStatusData) {
processJobStatusData(data)
}
```
I think it would improve readability

Reply:
it's maybe a bit messy right now but in  pull request #147 each if block is just one line